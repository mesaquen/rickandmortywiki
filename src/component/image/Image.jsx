import React, { useState } from 'react';
import Skeleton from 'react-loading-skeleton';
import styled from 'styled-components';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-11-16
 */
const StyledImage = styled.img`
  display: ${props => (props.ready ? 'block' : 'none')};
`;

const Image = props => {
  const [ready, setReady] = useState(false);

  const handleLoad = () => void setReady(true);
  const { width, height, src } = props;

  return (
    <>
      {ready ? null : <Skeleton ready={ready} width={width} height={height} />}
      <StyledImage
        ready={ready}
        src={src}
        onLoad={handleLoad}
        width={width}
        height={height}
      />
    </>
  );
};

export default React.memo(Image);
