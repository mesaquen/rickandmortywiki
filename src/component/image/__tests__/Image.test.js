import React from 'react';
import { render } from '@testing-library/react';
import Image from '../Image';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-11-16
 */

describe('Image', () => {
  const { asFragment } = render(<Image src="none.jpg" />);
  it('should have same snapshot when loading', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
