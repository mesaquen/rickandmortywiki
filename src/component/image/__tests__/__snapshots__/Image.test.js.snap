// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should have same snapshot when hidden 1`] = `
Array [
  <span>
    <span
      className="css-19bon8n react-loading-skeleton"
      style={
        Object {
          "animation": "animation-16jpnkj 1.2s ease-in-out infinite",
        }
      }
    >
      ‌
    </span>
  </span>,
  <img
    className="sc-bdVaJa fGYvF"
    onLoad={[Function]}
    src="none.jpg"
  />,
]
`;

exports[`Modal should have same snapshot when loading 1`] = `
Array [
  <span>
    <span
      className="css-19bon8n react-loading-skeleton"
      style={
        Object {
          "animation": "animation-16jpnkj 1.2s ease-in-out infinite",
        }
      }
    >
      ‌
    </span>
  </span>,
  <img
    className="sc-bdVaJa fGYvF"
    onLoad={[Function]}
    src="none.jpg"
  />,
]
`;
